halo2_backend/src/poly/kzg/strategy.rs

```diff
@@ -155,7 +155,7 @@ where
     E::G1: CurveExt<AffineExt = E::G1Affine>,
     E::G2Affine: SerdeCurveAffine,
 {
-    type Output = ();
```

Comment:
I see that all the implementations of `VerificationStrategy` have `Output = Self`.
That seems redundant, we could remove this associated type.  Then the verification process either returns bool, or the `VerificationStrategy` object, no need to do `VerificationStrategy::Output` which is already `VerificationStragey` all the time.

Also, I think we should discourage the usage of verification functions that return the `VerificationStrategy` because returning a `bool` is much clearer.  Since this PR is scheduled to be merged after the `0.4` release, and we determined that in the `0.5` release we can remove the `halo2_proofs` legacy layer, maybe we can just remove the verification methods that return `VerificationStrategy`?  If not, at least could we mark them as [deprecated via `cfg`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute)?

Reply:
I like the proposal from @ed255 
And if we break API is indeed a good time to do it!

Reply:
@ed255 
I agree that `Output = Self` is redundant.
For the quick patch, I remove the `Output` type from `VerificationStrategy` trait. https://github.com/privacy-scaling-explorations/halo2/pull/359/commits/e5153c4e57d76d9ace3fa951c185ff22b78dd672

> Or is there any strong reason to keep the API that takes and returns a VerificationStrategy object?

This is related to `process` API.
I think this API could be useful for batch verification, like following one.

https://github.com/privacy-scaling-explorations/halo2/blob/9fc59c100ed3e65f58c174853f8bd4d6c2bc8231/halo2_backend/src/plonk/verifier/batch.rs#L93-L129
The `BatchVerifier` delays the MSM check until it accumulates all the MSM from every batch items.

What do you think about this? @ed255 @CPerezz

Reply:
The current state of this PR looks good to me (removal of `Output` from trait).